### PR TITLE
chore(refactor): Refactor core code

### PR DESCRIPTION
**Pull request by roneli**

### CODE REVIEW

File: cmd/command.go [hunk 1]
```diff
@@ -97,24 +68,22 @@ func handleConsole(ctx context.Context, tele *telemetry.Client, cmd *cobra.Comma
...
-		c, err = console.CreateClient(ctx, cfgPath, cfgMutator)
+		c, err = console.CreateClient(ctx, cfgPath, false, cfgMutator)
 		if err != nil {
 			return err
 		}
-		defer c.Client().Close()
 	}
 
-	if tele.Enabled() && tele.NewRandomId() {
+	if analytics.Enabled() {
```
Comment (disq) [inline cmd/command.go:86]: There's a functionality disparity here: we only do this notice/delay thing if we had created a "new" ID, so it was only shown once (unless the filesystem isn't persistent)

New code shows/delays every time

File: cmd/provider.go [hunk 2]
```diff
@@ -42,7 +46,12 @@ var (
 		Short: providerUpgradeHelpMsg,
 		Long:  providerUpgradeHelpMsg,
 		Run: handleCommand(func(ctx context.Context, c *console.Client, cmd *cobra.Command, args []string) error {
-			return c.UpgradeProviders(ctx, args)
+			_, diags := c.SyncProviders(ctx, args...)
+			errors.CaptureDiagnostics(diags, nil)
```
Comment (disq) [inline cmd/provider.go:50]: Missing command in the map

File: cmd/provider.go [hunk 3]
```diff
@@ -52,7 +61,12 @@ var (
 		Short: providerDowngradeHelpMsg,
 		Long:  providerDowngradeHelpMsg,
 		Run: handleCommand(func(ctx context.Context, c *console.Client, cmd *cobra.Command, args []string) error {
-			return c.DowngradeProviders(ctx, args)
+			_, diags := c.SyncProviders(ctx, args...)
+			errors.CaptureDiagnostics(diags, nil)
```
Comment (disq) [inline cmd/provider.go:65]: Missing command in the map

File: cmd/command.go (hunk 1 above)
Comment (roneli) [reply cmd/command.go:86]: I think it should show every time to be honest even if Id is already created.

Comment (disq) [reply cmd/command.go:86]: but it also delays for 2 seconds.

Comment (roneli) [reply cmd/command.go:84]: we can live with that for now ?

**Pull request merged**